obsreport/obsreporttest/otelprometheuschecker.go

```diff
@@ -34,6 +34,13 @@ type prometheusChecker struct {
 	promHandler http.Handler
 }
 
+func (pc *prometheusChecker) checkScraperMetrics(receiver config.ComponentID, scraper config.ComponentID, scrapedMetricPoints, erroredMetricPoints int64) error{
```

Comment:
Please rebase and add tests for these as well.

Reply:
Thanks for the review! Done!